src/client/opamCommands.ml

```diff
@@ -3184,6 +3184,10 @@ let pin ?(unpin_only=false) cli =
        changes, and may also be used to keep a package that was removed \
        upstream."
   in
+  let all =
+    mk_flag ~cli (cli_from cli2_2) ["all"]
+      "When unpinning, removes all pins in the given switch."
```

Comment:
Should we add a word about this command being a no-op on other pin commands ?

Reply:
I chose to make the use of `--all` outside of `remove` without any parameter an error instead. Does that work for you?

Reply:
good for me!